src/compiler/csharp_generator.cc

```diff
@@ -323,8 +323,52 @@ std::vector<const Descriptor*> GetUsedMessages(
   return result;
 }
 
-void GenerateMarshallerFields(Printer* out, const ServiceDescriptor* service) {
+void GenerateMarshallerFields(Printer* out, const ServiceDescriptor* service,
+                              bool use_buffer_serialization) {
   std::vector<const Descriptor*> used_messages = GetUsedMessages(service);
+  out->Print("#if !GOOGLE_PROTOBUF_DISABLE_BUFFER_SERIALIZATION\n");
```

Comment:
Exit this method without writing anything if used_messages is empty. That avoids empty `#if` when there are no messages